server/src/main/java/org/elasticsearch/cluster/routing/allocation/RoutingAllocation.java

```diff
@@ -92,16 +94,18 @@ public RoutingAllocation(
      * @param clusterInfo information about node disk usage and shard disk usage
      * @param shardSizeInfo information about snapshot shard sizes
      * @param currentNanoTime the nano time to use for all delay allocation calculation (typically {@link System#nanoTime()})
+     * @param reason for the allocation cycle
      */
     public RoutingAllocation(
         AllocationDeciders deciders,
         @Nullable RoutingNodes routingNodes,
         ClusterState clusterState,
         ClusterInfo clusterInfo,
         SnapshotShardSizeInfo shardSizeInfo,
-        long currentNanoTime
+        long currentNanoTime,
+        @Nullable String reason
```

Comment:
Does it really need to be nullable? I only see a few ways it can be `null` in production code and all of them really could have a reason instead.